src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -131,99 +130,99 @@ writeSys wt d k v = do
         Namespaces -> convNamespaceName
         Pacts -> convPactId
         UserTables _ -> error "impossible"
-    write key bh version txid db =
-        let f = case wt of
-                    Insert -> backendWriteInsert
-                    Update -> backendWriteUpdate
-                    Write -> backendWriteWrite
-        in f key (domainTableName d) bh version txid v db
+
+    write key bh txid db = f key (domainTableName d) bh txid v db
+      where
+        f = case wt of
+            Insert -> backendWriteInsert
+            Update -> backendWriteUpdate
+            Write -> backendWriteWrite
 
 backendWriteInsert
     :: ToJSON v
     => Utf8
     -> Utf8
     -> BlockHeight
-    -> ReorgVersion
     -> TxId
     -> v
     -> Database
     -> IO ()
-backendWriteInsert key tn bh version txid v db =
+backendWriteInsert key tn bh txid v db = do
     exec' db q [ SText key
                , SInt (fromIntegral bh)
-               , SInt (fromIntegral version)
                , SInt (fromIntegral txid)
                , SBlob (toStrict (Data.Aeson.encode v))]
+    exec' db mutq [SText tn, SInt (fromIntegral bh)]
   where
+    mutq = "INSERT OR IGNORE INTO VersionedTableMutation VALUES (?,?);"
```

Comment:
note for possible performance followup -- this is a place where a per-block cache would be appropriate, so we don't keep ramming sqlite

Reply:
understood. I'll keep this in mind.